x-pack/plugins/infra/public/store/local/log_flyout/reducer.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { combineReducers } from 'redux';
+import { reducerWithInitialState } from 'typescript-fsa-reducers';
+
+import { setFlyoutItem, showFlyout } from './actions';
+
+export interface FlyoutOptionsState {
+  isFlyoutVisible: boolean | null;
+  flyoutId: string | null;
```

Comment:
Similarly, to make forbidden state impossible to represent, we could write this as something like

```typescript

interface HiddenFlyout {
  kind: 'hidden';
}

interface VisibleFlyout {
  kind: 'visible';
  itemId: string;
}

export interface FlyoutOptionsState {
  flyoutVisibility: HiddenFlyout | VisibleFlyout;
}
```

That way, a visible flyout is always accompanied by an id.

Reply:
I resolve this with a similar solution but keeps with the spirit of the comment